.github/workflows/ci-package.yml

```diff
@@ -562,7 +562,7 @@ jobs:
         prerelease: false
 
     - name: Download Linux
-      uses: actions/download-artifact@v4.1.7
```

Comment:
Wait, you're downgrading this?  That's not in the PR description.  The security advisory https://github.com/advisories/GHSA-cxww-7g56-2vh6 says we need 4.1.7+: "Versions of actions/download-artifact before 4.1.7 are vulnerable to arbitrary file write".

Oh wait: it also says "Upgrade to version 4.1.7 or higher. Alternatively use 'v4' tag which points to the latest and secure version.".  That is confusing: "v4" doesn't mean 4.0?

This maybe helps clarify, saying "v4" will point to the latest only if the action maintainers took steps to make that happen: https://stackoverflow.com/questions/74962405/how-does-github-actions-workflow-resolve-minor-versions-of-an-actions

Could you confirm you see 4.1.7+ being used in the logs?  I looked and I see "Run actions/download-artifact@v4" vs the broken one "Run actions/download-artifact@v4.1.7": which maybe doesn't disprove it's 4.1.7.  But it would be good to know for sure.

Does "v4" for upload work with "v4.1.7" for download?  That feels the safest?

Reply:
> Alternatively use 'v4' tag which points to the latest and secure version.

Doesn't this statement in the security advisory already mean that "v4" is what we want?

Additionally, the many examples on upload-artifact and download-artifact action repo README suggests using v4 (mentioned in the PR description), so I assumed that's the canonical way.

> I looked and I see "Run actions/download-artifact@v4" vs the broken one "Run actions/download-artifact@v4.1.7":

Yeah that may just be parroting the "vX" string in the GA workflow config.

Reply:
Found more documentation:

https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsuses says "If the action publishes major version tags, you should expect to receive critical fixes and security patches while still retaining compatibility. Note that this behavior is at the discretion of the action's author." I do see "v4" at https://github.com/actions/upload-artifact/tags, which has the same commit has as "v4.4.0" (5076954), so it is indeed pointed to the latest?

Reply:
Same for https://github.com/actions/download-artifact/tags: "v4" is pointed at "v4.1.8" (fa0a91b).

Reply:
Hmm... if we use "v4" on both, that's v4.4.0 and v.4.1.8 on the two actions. Isn't that a version mismatch? Perhaps "real" mismatches would be between different major versions? I guess we're trusting the maintainers of upload-artifact and download-artifact to make sure "v4" on both are compatible with each other. I think that's fine.

Reply:
If you could add a comment briefly explaining that this is not v4.0.0 but in fact the latest 4.x.x that will avoid confusion in the future (when future me forgets, or a different dev looks at this).  Putting the same thing in the PR description would help too.

Reply:
Already added to the PR description. Added to a comment also now.